src/App.js

```diff
@@ -1,16 +1,41 @@
-import React from 'react';
+import React, {useState} from 'react';
 import './App.css';
-import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+import HeartCounter from './components/HeartCounter'
+import messagesJSON from './data/messages.json'
 
 const App = () => {
+  const [messageData, setMessageData] = useState(messagesJSON);
+  const [heartTotal, setHeartTotal] = useState(0);
+
+  const toggleLiked = (id) => {
+    const messages = messageData.map((message) => {
+      if (message.id === id) {
+        if (message.liked === false) {
+          setHeartTotal(heartTotal + 1);
+        } else {
+          setHeartTotal(heartTotal -1);
+        };
+
+        message.liked = !message.liked;
```

Comment:
Here, you're mutating the original message data that's imported on line 5 above. This was actually causing some tests to fail, in conjunction with a few other points mentioned above and below. For React, you'll want to be cautious about such mutations as it can cause unexpected bugs. The best approach to updating data is actually to create new objects, e.g.
```js
  const toggleLiked = (id) => {
    const messages = messageData.map((message) => {
      if (message.id === id) {
        if (message.liked === false) {
          setHeartTotal((prevHeartTotal) => prevHeartTotal + 1);
        } else {
          setHeartTotal((prevHeartTotal) => prevHeartTotal - 1);
        };

        return {
          ...message,
          liked: !message.liked,
        }

      } else {
        return message;
      }
    });

    setMessageData(messages)
}
```
But better still would be to do this while also leveraging the approach outlined above of reading the previous data passed to a callback function of the state updating function, e.g.
```js
  const toggleLiked = (id) => {
    setMessageData((prevMessages => {
      return prevMessages.map((message) => {
        if (message.id === id) {
          return {
            ...message,
            liked: !message.liked,
          }
        } else {
          return message;
        }
      });
    }));
  }
```
and then calculating the heart total separately
```js
  const heartTotal = messageData.filter(m => m.liked).length;
```